runtime/stdlib/builtin.go

```diff
@@ -179,6 +181,130 @@ var CreatePublicKeyFunction = NewStandardLibraryFunction(
+	func(invocation interpreter.Invocation) interpreter.Value {
+		signatures := invocation.Arguments[0].(*interpreter.ArrayValue)
+		return AggregateBLSSignatures(invocation.Interpreter, signatures)
+	},
+)
+
+const aggregateBLSPublicKeysFunctionDocString = `
+This is a specific function for the BLS signature scheme. 
+It aggregates multiple BLS public keys into one.
+
+The order of the public keys in the slice does not matter since the aggregation is commutative. 
+No subgroup membership check is performed on the input keys.
+The function errors if the array is empty or any of the input keys is not a BLS key.
+`
+
+var AggregateBLSPublicKeysFunction = NewStandardLibraryFunction(
+	"AggregateBLSPublicKeys",
+	&sema.FunctionType{
+		Parameters: []*sema.Parameter{
+			{
+				Label:          sema.ArgumentLabelNotRequired,
+				Identifier:     "keys",
+				TypeAnnotation: sema.NewTypeAnnotation(&sema.VariableSizedType{Type: sema.PublicKeyType}),
+			},
+		},
+		ReturnTypeAnnotation: sema.NewTypeAnnotation(sema.PublicKeyType),
+	},
+	aggregateBLSPublicKeysFunctionDocString,
+	func(invocation interpreter.Invocation) interpreter.Value {
+		publicKeys := invocation.Arguments[0].(*interpreter.ArrayValue)
+		return AggregateBLSPublicKeys(
+			invocation.Interpreter,
+			invocation.GetLocationRange,
+			publicKeys,
+		)
+	},
+)
+
+func AggregateBLSPublicKeys(
+	inter *interpreter.Interpreter,
+	getLocationRange func() interpreter.LocationRange,
+	publicKeys *interpreter.ArrayValue,
+) interpreter.Value {
+	publicKeyArray := make([]interpreter.MemberAccessibleValue, 0, publicKeys.Count())
+	publicKeys.Iterate(func(element interpreter.Value) (resume bool) {
+		publicKey := element.(*interpreter.CompositeValue)
+		publicKeyArray = append(publicKeyArray, publicKey)
+		return true
+	})
+
+	aggregatedKey, err := inter.AggregateBLSPublicKeysHandler(
+		inter,
+		getLocationRange,
+		publicKeyArray,
+	)
+	if err != nil {
+		panic(err)
+	}
+
+	return aggregatedKey
+}
+
+func AggregateBLSSignatures(
+	inter *interpreter.Interpreter,
+	signatures *interpreter.ArrayValue,
+) interpreter.Value {
+	bytesArray := make([][]byte, 0, signatures.Count())
+	signatures.Iterate(func(element interpreter.Value) (resume bool) {
+		sig := element.(*interpreter.ArrayValue)
+		bytes := make([]byte, 0, sig.Count())
+		sig.Iterate(func(element interpreter.Value) (resume bool) {
+			i := element.(interpreter.UInt8Value)
+			bytes = append(bytes, byte(i))
+			return true
+		})
+		bytesArray = append(bytesArray, bytes)
+		return true
+	})
+
+	aggregatedBytes, err := inter.AggregateBLSSignaturesHandler(
+		bytesArray,
+	)
+	if err != nil {
+		panic(err)
+	}
+
+	aggregatedSignature := make([]interpreter.Value, 0, len(aggregatedBytes))
+	for _, b := range aggregatedBytes {
+		aggregatedSignature = append(aggregatedSignature, interpreter.UInt8Value(b))
+	}
+
+	return interpreter.NewArrayValue(
+		inter,
+		interpreter.VariableSizedStaticType{Type: interpreter.PrimitiveStaticTypeUInt8},
```

Comment:
Maybe use the existing constant for this:

```suggestion
		interpreter.ByteArrayStaticType,
```